### ISSUE TITLE
Capture ourselves the STDOUT and STDERR of ClusterTasks

### ISSUE THREAD

**Issue by prioux** [aces/cbrain#133]
#### Main feature

On the clusters, instead of showing the STDOUT and STDERR as captured by the qsub/msub commands, which often are not sent to the files in real time, we should introduce a new layer of wrapping that captures them to file ourselves. That way, when looking at a task that is currently "On CPU", we would be able to see the outputs as they are being produced. The bash script submitted to the cluster would be simple and  look something like:

```bash
#!/bin/bash

bash original_bash_script 1>outcapture 2>errcapture
```

#### Special supplemental feature

Since often the cluster's management system also produces useful prologue and epilogue associated with the submitted wrapper, we could integrate merge the full thing with a special keyword placeholder:

```bash
#!/bin/bash

echo _SPECIAL_PLACEHOLDER_FOR_STDOUT_SUBSITUTION_
echo _SPECIAL_PLACEHOLDER_FOR_STDERR_SUBSITUTION_ 1>&2
bash original_bash_script 1>outcapture 2>errcapture
```

In that case, after the job is fully finished, we can rewrite the content of outcapture and errcapture to include the prologue and epilogue outputs.

**Comment by prioux** [comment]
I really need to think about this carefully.

**Comment by prioux** [comment]
Related to issue #64 : both can be done at the same time.

**Comment by glatard** [comment]
I'm tempted to argue for the exit code check again here. If the application commands are wrapped in a separate script, then I really think it would make sense that the exit code of this script can be retrieved, and the task would be failed on cluster when the exit code is not 0 *and a specific task property, e.g. :check_exit_code_value, has been set by the task developer*. When this property is activated, it is then up to the task developer to write a script that will properly set the exit code when the task fails. This would prevent task developers from writing their own buggy exit code checks in an ad-hoc manner.

**Comment by glatard** [comment]
This feature would be very useful for the PSOM integration as PSOM pipelines report on their progression during execution (using #364).

**Comment by glatard** [comment]
This would be really nice to have for long processes like PSOM pipelines that write progress information in their stdout during execution.

**Comment by poquirion** [comment]
A feature I have seen in a some cluster manager I have used il the past is the possibility to provide a path to be feed tou a `tail` like function and/our a path plus a regex to feed a number to a progress bar. In any case a feature to seen what is going with the computation is more than necessary for the cbrain psom integration. I have a test that have been running for two days, and I have no feed back except for the fact that  it in `On CPU`.

**Comment by prioux** [comment]
Fixed by #524.
